src/vm/eventpipe.cpp

```diff
@@ -293,6 +301,9 @@ EventPipeSessionID EventPipe::Enable(
 
     EventPipeSessionID sessionId = 0;
     RunWithCallbackPostponed([&](EventPipeProviderCallbackDataQueue *pEventPipeProviderCallbackDataQueue) {
+        if (!s_tracingInitialized)
```

Comment:
Just curious: What does moving this check inside the callback change?

Reply:
Inside the callback, the check is protected by the Crst.